src/todolist.js

```diff
@@ -1 +1,53 @@
+class TodoList {
+  constructor() {
+    this.id = 0
+    this.items = []
+  }
 
+  create(todo) {
+    this.id++
+    const todoItem = {
+      id: this.id,
+      description: todo,
+      status: 'incomplete'
+    }
+    this.items.push(todoItem)
+    return todoItem
+  }
+
+  getItems() {
+    return this.items
+  }
+
+  completeTodo(id) {
+    if (this.items[id - 1]) {
+      this.items.find((item) => item.id === id).status = 'completed'
+      return this.items[id - 1]
+    }
+    return 'Todo item not found'
+  }
+
+  getIncompleteTodos() {
+    return this.items.filter((item) => item.status === 'incomplete')
+  }
+
+  getCompleteTodos() {
+    return this.items.filter((item) => item.status === 'completed')
+  }
+
+  searchId(id) {
+    if (this.items[id - 1]) {
+      return this.items[id - 1]
+    }
+    return 'Does not exist'
+  }
+
+  deleteId(id) {
+    if (this.items[id - 1]) {
+      return this.items.splice(id - 1, 1)
+    }
+    return 'Does not exist'
```

Comment:
We haven't covered it yet but instead of returning a string, you could throw an error message.

```js
throw 'Does not exist'
```

In source code usage of that function then, you'd have to wrap calling the function inside a try / catch block (I'm happy to cover this in a lesson if it's new):

```js
try {
  todoList.delete(2)
} catch (e) {
  console.error("An error happened...", e)
}
```

To create a unit test in jasmine for things that throw errors, you can use the `toThrow` matcher. This one's a little different, you'll need to give the `expect` a callback function it can run, like this:

```js
expect(() => todoList.delete(2)).toThrow()
```